EventFilter/Utilities/plugins/RecoEventOutputModuleForFU.h

```diff
@@ -304,7 +320,15 @@ namespace evf {
       //lock
       struct stat istat;
       if (!edm::Service<evf::EvFDaqDirector>()->microMergeDisabled()) {
-        FILE *des = edm::Service<evf::EvFDaqDirector>()->maybeCreateAndLockFileHeadForStream(ls.luminosityBlock(),stream_label_);
+
+        //create if does not exist then lock the merge destination file
+        FILE *des = fopen(edm::Service<evf::EvFDaqDirector>()->getMergedDatFilePath(ls.luminosityBlock(),stream_label_).c_str(), "a"); //open stream for appending
+        int data_readwrite_fd_ = fileno(des);
```

Comment:
Please remove the trailing underscore on the variable since CMS convention is trailing underscore is used for member data.

Reply:
Seems like using RAII could be good here
```C++
auto finishFile = [data_readwrite_fd_, this](FILE* f) { 
   fflush(f);
   fcntl(data_readwrite_fd_,F_SETLKW, &data_rw_fulk);
   fclose(des);
};

std::unique_ptr<FILE,decltype(finishFile)> desGuard{des,finishFile};
```

Then lines 375-379 go away since the destructor of the `std::unique_ptr` will do the work. This will even guarantee that `finishFile` will be called if there is an exception thrown. (such as is done on line 342 already).